src/detect-dce-iface.c

```diff
@@ -180,8 +179,12 @@ static DetectDceIfaceData *DetectDceIfaceArgParse(const char *arg)
             goto error;
         }
 
-        version = atoi(copy_str);
-        if (version > UINT16_MAX) {
+        if (ByteExtractStringInt32(&version, 10, 0, (const char *)copy_str) < 0) {
```

Comment:
Cannot we do `ByteExtractStringUInt16`instead of `ByteExtractStringInt32` ? and save the letter check ?

Reply:
Yes, we could. I'll declare version as `uint16_t` then else the compiler will throw warnings.

Reply:
If we could, then I guess we should :-)